src/main/java/htsjdk/variant/variantcontext/writer/VCFWriter.java

```diff
@@ -128,13 +131,14 @@ private void writeAndResetBuffer() throws IOException {
 
     @Override
     public void writeHeader(final VCFHeader header) {
+
         // note we need to update the mHeader object after this call because they header
         // may have genotypes trimmed out of it, if doNotWriteGenotypes is true
         try {
-            this.mHeader = writeHeader(header, writer, doNotWriteGenotypes, getVersionLine(), getStreamName());
-            this.vcfEncoder = new VCFEncoder(this.mHeader, this.allowMissingFieldsInHeader, this.writeFullFormatField);
+            setVCFHeader(header);
+            writeHeader(header, writer, getVersionLine(), getStreamName());
```

Comment:
@amilamanoj This one also is using the stale (local) header. It should use the member variable.